Source/SwiftLintFramework/Rules/Idiomatic/RedundantSetAccessControlRule.swift

```diff
@@ -46,10 +46,50 @@ public struct RedundantSetAccessControlRule: ASTRule, ConfigurationProviderRule,
         ]
     )
 
-    public func validate(file: File, kind: SwiftDeclarationKind,
-                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+    public func validate(file: File) -> [StyleViolation] {
+        return validate(file: file, dictionary: file.structure.dictionary, parentDictionary: nil)
+    }
+
+    private func validate(file: File, dictionary: [String: SourceKitRepresentable],
+                          parentDictionary: [String: SourceKitRepresentable]?) -> [StyleViolation] {
+        return dictionary.substructure.flatMap { subDict -> [StyleViolation] in
+            var violations = validate(file: file, dictionary: subDict, parentDictionary: dictionary)
+
+            if let kindString = subDict.kind,
+                let kind = SwiftDeclarationKind(rawValue: kindString) {
+                violations += validate(file: file, kind: kind, dictionary: subDict, parentDictionary: dictionary)
+            }
+
+            return violations
+        }
+    }
+
+    private func validate(file: File, kind: SwiftDeclarationKind,
+                          dictionary: [String: SourceKitRepresentable],
+                          parentDictionary: [String: SourceKitRepresentable]?) -> [StyleViolation] {
+        let aclAttributes: Set<SwiftDeclarationAttributeKind> = [.private, .fileprivate, .internal, .public, .open]
```

Comment:
would love some 👀 on this change

Reply:
I'm not sure if the changes to violations reported by OSSCheck are from this change or Swift 5 or both.